### DISCUSSION TITLE
Better console logging of fetch caching state on server side fetch call

### DISCUSSION THREAD

**Discussion by InTheNooB** [vercel/next.js#54373] [Ideas]
### Goals

1. Better understanding of the state of the fetch cache
2. Less time spend controlling that cache works
3. Better productivity

### Non-Goals

1.
2.
3.

### Background

The situation is :
- An app runs in DEV
- A request is sent to an api route
- This api route sends a request to a service using fetch
- This fetch call has next.revalidate = XYZ set

Whenever the request is sent to the api route, some information about this request is written in the console output :
```
-  ┌ GET /api/whatever 200 in 13ms
   │
   └──── GET https://random-data-api.com/api/v2/users 200 in 1ms (cache: HIT)
```

The `(cache: HIT)` statement here is misleading. Let me explain: 
- On the first request sent, it says `(cache: MISS)` **(normal behavior)**
- Every following requests sent in between the next.revalidate time will say `(cache: HIT)` **(normal behavior)**
- But the request right after the next.revalidate time is over, which is then actually sent to `random-data-api.com` will also say `(cache: HIT)`, which is **WRONG** because the cache was stall and was just updated, so there was no `HIT`.

### Proposal

The `(cache: STATE)` should be something like `(cache: REVALIDATED)` instead of always being `(cache: HIT)` when a fetch call has been revalidated.

I don't think this is too much work, so I guess I could work on it, but I have no clue where to start.

**Comment by InTheNooB** [comment]
This might be a bug now that I think about it

**Reply by XEngine** [reply]
I have another question related to this. After 13.4.12+ they seem to remove this output from console. Are you using a latest version or is there a way to enable this logging again?

**Reply by InTheNooB** [reply]
Yeah you are right, they removed it. I don't understand why tho because it was really helpful to better understand what is happening
